File: java/src/Ice/src/main/java/com/zeroc/Ice/ProtocolInstance.java
//
// Copyright (c) ZeroC, Inc. All rights reserved.
//

package com.zeroc.Ice;

/**
 * @hidden Public because it's used by IceBT, SSL, and the 'Ice/background' test.
 */
public class ProtocolInstance {
    public ProtocolInstance(
            Communicator communicator, short type, String protocol, boolean secure) {
        _instance = communicator.getInstance();
        _traceLevel = _instance.traceLevels().network;
        _traceCategory = _instance.traceLevels().networkCat;
        _logger = _instance.initializationData().logger;
        _properties = _instance.initializationData().properties;
        _type = type;
        _protocol = protocol;
        _secure = secure;
    }

    public int traceLevel() {
        return _traceLevel;
    }

    public String traceCategory() {
        return _traceCategory;
    }

    public Logger logger() {
        return _logger;
    }

    public EndpointFactory getEndpointFactory(short type) {
        return _instance.endpointFactoryManager().get(type);
    }

    public String protocol() {
        return _protocol;
    }

    public short type() {
        return _type;
    }

    public boolean secure() {
        return _secure;
    }

    public Properties properties() {
        return _properties;
    }

    public boolean preferIPv6() {
        return _instance.preferIPv6();
    }

    public int protocolSupport() {
        return _instance.protocolSupport();
    }

    public String defaultHost() {
        return _instance.defaultsAndOverrides().defaultHost;
    }

    public java.net.InetSocketAddress defaultSourceAddress() {
        return _instance.defaultsAndOverrides().defaultSourceAddress;
    }

    public EncodingVersion defaultEncoding() {
        return _instance.defaultsAndOverrides().defaultEncoding;
    }

    public NetworkProxy networkProxy() {
        return _instance.networkProxy();
    }

    public int messageSizeMax() {
        return _instance.messageSizeMax();
    }

    public void resolve(
            String host,
            int port,
            EndpointSelectionType type,
            IPEndpointI endpt,
            EndpointI_connectors callback) {
        _instance.endpointHostResolver().resolve(host, port, type, endpt, callback);
    }

    public BufSizeWarnInfo getBufSizeWarn(short type) {
        return _instance.getBufSizeWarn(type);
    }

    public void setSndBufSizeWarn(short type, int size) {
        _instance.setSndBufSizeWarn(type, size);
    }

    public void setRcvBufSizeWarn(short type, int size) {
        _instance.setRcvBufSizeWarn(type, size);
    }

    ProtocolInstance(Instance instance, short type, String protocol, boolean secure) {
        _instance = instance;
        _traceLevel = _instance.traceLevels().network;
        _traceCategory = _instance.traceLevels().networkCat;
        _logger = _instance.initializationData().logger;
        _properties = _instance.initializationData().properties;
        _type = type;
        _protocol = protocol;
        _secure = secure;
    }

    protected final Instance _instance;
    protected int _traceLevel;
    protected String _traceCategory;
    protected Logger _logger;
    protected Properties _properties;
    protected String _protocol;
    protected short _type;
    protected boolean _secure;
}

File: csharp/src/Ice/Internal/ProtocolInstance.cs
// Copyright (c) ZeroC, Inc.

using System.Net;

namespace Ice.Internal;

public class ProtocolInstance
{
    public ProtocolInstance(Ice.Communicator communicator, short type, string protocol, bool secure)
    {
        instance_ = communicator.instance;
        traceLevel_ = instance_.traceLevels().network;
        traceCategory_ = instance_.traceLevels().networkCat;
        logger_ = instance_.initializationData().logger;
        properties_ = instance_.initializationData().properties;
        type_ = type;
        protocol_ = protocol;
        secure_ = secure;
    }

    public ProtocolInstance(Instance instance, short type, string protocol, bool secure)
    {
        instance_ = instance;
        traceLevel_ = instance_.traceLevels().network;
        traceCategory_ = instance_.traceLevels().networkCat;
        logger_ = instance_.initializationData().logger;
        properties_ = instance_.initializationData().properties;
        type_ = type;
        protocol_ = protocol;
        secure_ = secure;
    }

    public int traceLevel() => traceLevel_;

    public string traceCategory() => traceCategory_;

    public Ice.Logger logger() => logger_;

    public EndpointFactory getEndpointFactory(short type) => instance_.endpointFactoryManager().get(type);

    public string protocol() => protocol_;

    public short type() => type_;

    public bool secure() => secure_;

    public Ice.Properties properties() => properties_;

    public bool preferIPv6() => instance_.preferIPv6();

    public int protocolSupport() => instance_.protocolSupport();

    public string defaultHost() => instance_.defaultsAndOverrides().defaultHost;

    public EndPoint defaultSourceAddress() => instance_.defaultsAndOverrides().defaultSourceAddress;

    public Ice.EncodingVersion defaultEncoding() => instance_.defaultsAndOverrides().defaultEncoding;

    public NetworkProxy networkProxy() => instance_.networkProxy();

    public int messageSizeMax() => instance_.messageSizeMax();

    public void resolve(string host, int port, IPEndpointI endpt, EndpointI_connectors callback) =>
        instance_.endpointHostResolver().resolve(host, port, endpt, callback);

    public BufSizeWarnInfo getBufSizeWarn(short type) => instance_.getBufSizeWarn(type);

    public void setSndBufSizeWarn(short type, int size) => instance_.setSndBufSizeWarn(type, size);

    public void setRcvBufSizeWarn(short type, int size) => instance_.setRcvBufSizeWarn(type, size);

    protected readonly Instance instance_;
    protected int traceLevel_;
    protected string traceCategory_;
    protected Ice.Logger logger_;
    protected Ice.Properties properties_;
    protected string protocol_;
    protected short type_;
    protected bool secure_;
}
